Tests/add event listeners

TxHawks opened the pull request:
his pull request adds tests to `addEventListeners()`, as well as fixes a bug found in the code.

TxHawks committed:
[WIP] addEventListeners tests

TxHawks committed:
Add addEventListeners tests

nadavsinai committed:
fix the add eventlistener spec - actual source code bug...

TxHawks committed:
Fix passing undefined instead of elem to keepFocus

`keepFocus` was passed `wrapper`, which is `undefined`, instead of
`dialofState.wrapper` which is an `HTMLElement`.

TxHawks committed:
Clean up inserted DOM elements after test

TxHawks committed:
Fix linting errors related to addEventListreners

nadavsinai commented on .eslintrc.js:
```diff
@@ -8,9 +8,15 @@ module.exports = {
+  },
+
+  "globals": {
+    "expect": true,
```
this is not needed, I inject the  globals via webpack eslint loader - see test-web and test-node configs
setting this here will mean that eslint will think you're in mocha all the time - also in dev mode...

TxHawks replied:
If I remove this I get `warning  'sinon' is not defined` and `warning  'expect' is not defined` when running `npm run lint`
```

nadavsinai commented on src/lib/addEventListeners.spec.js:
```diff
@@ -1,19 +1,221 @@
 
-  it('adds onKeyDown listener to the wrapper for keyCode 27 if element is visible');
+  describe('Keydown events on the wrapper', () => {
+    it('adds keydown listener to the wrapper', () => {
```
well, you're right this is brittle. let's remove this test
we can just check behaviour, as I said, strictly speaking this is not completely a unit test, but it does not cross to internal implementation which is better than the current test, 
just perhaps rephrase the test below  to be:
it('makes a visible dialog hidden using hide() when pressing 'esc')

nadavsinai commented on src/lib/addEventListeners.spec.js:
```diff
@@ -1,19 +1,221 @@
+      wrapper.remove();
+    });
+
+    it('Does not execute keepFocus on keydown when keyCode is 9 & wrapper is *NOT* visible', () => {
```
for the sake of DRYness we can make these assertions within the test above using the same setup and cleanup or abstract these and only change the keyCode and assertion

nadavsinai commented:
lets just revert the eslint globals then OK to merge

TxHawks committed:
Remove brittle test

The removed test was tightly coupled to implementation
and didn't test actual behaviour.

TxHawks committed:
Dry up tests with prep function

TxHawks committed:
Clearer test descriptions

TxHawks committed:
Make test code even drier

TxHawks committed:
fix linting errors in spec file

TxHawks commented:
All comments addressed, except the one about changes in `.eslintrc.js` (See comment [here](https://github.com/nadavsinai/htz-dialog/pull/7/files#r97880639))

nadavsinai committed:
Merge remote-tracking branch 'jwhaks/tests/addEventListeners' into tests/addEventListeners

# Conflicts:
#	src/lib/addEventListeners.js
#	src/lib/addEventListeners.spec.js

nadavsinai committed:
fix npm run lint to use the test-globals without making those globals be defined in devmode lint (currently disabled)

nadavsinai committed:
another fix for the npm run lint globals
